src/lib.rs

```diff
@@ -75,13 +138,46 @@ pub struct QueueData {
     pub txn_count: BigDecimal,
 }
 
-#[derive(Deserialize)]
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(untagged)]
+pub enum LedgerIndex {
```

Comment:
Here we have a more complex type that we use the serde flatten and the serde untagged to pull out switching fields in the account requests.